test/functional/tool_wallet.py

```diff
@@ -545,6 +549,36 @@ def test_dump_unclean_lsns(self):
         self.stop_node(0)
         self.assert_tool_output("The dumpfile may contain private keys. To ensure the safety of your Bitcoin, do not share the dumpfile.\n", "-wallet=unclean_lsn", f"-dumpfile={wallet_dump}", "dump")
 
+    def test_compare_legacy_dump_with_framework_bdb_parser(self):
+        self.log.info("Verify that legacy wallet database dump matches the one from the test framework's BDB parser")
+        wallet_name = "bdb_ro_test"
+        self.start_node(0)
+        # add some really large labels (above twice the largest valid page size) to create BDB overflow pages
```

Comment:
In ad0a8ffceacfd5ce37826b2882b0e51a14c5a6bf "test: compare BDB dumps of test framework parser and wallet tool"

In addition to testing overflow pages, we can also test internal pages by generating a bunch of keys with `keypoolrefill`.